lib/internal/modules/esm/loader.js

```diff
@@ -70,28 +87,30 @@ class ESMLoader {
   /**
    * Prior to ESM loading. These are called once before any modules are started.
    * @private
-   * @property {Function[]} globalPreloaders First-in-first-out list of
-   * preload hooks.
+   * @property {Map<URL['href'], Function>} globalPreloaders Last-in-first-out
+   *  list of preload hooks.
    */
-  #globalPreloaders = [];
+  #globalPreloaders = new SafeMap();
 
   /**
    * Phase 2 of 2 in ESM loading.
    * @private
-   * @property {Function[]} loaders First-in-first-out list of loader hooks.
+   * @property {Map<URL['href'], Function>} loaders Last-in-first-out
+   *  collection of loader hooks.
    */
-  #loaders = [
-    defaultLoad,
-  ];
+  #loaders = new SafeMap([
+    ['node:esm/load.js', defaultLoad],
+  ]);
 
   /**
    * Phase 1 of 2 in ESM loading.
```

Comment:
I considered that back when I did the consolidation work. I opted for this because of Principle of Least Knowledge: alphabetical order doesn't require someone knowing about phases.